Source/Upstream: Yes, fixed in git master
Reason: Fix build with ffmpeg 8

From ee46e7c8e702f2d14687ae71f9fe49071da531be Mon Sep 17 00:00:00 2001
From: Paul Brossier <piem@piem.org>
Date: Sat, 15 Nov 2025 12:05:27 +0100
Subject: [PATCH] [source_avcodec] add backward compatible version of chlayout,
 thanks to @acrilique

---
 src/io/source_avcodec.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/io/source_avcodec.c b/src/io/source_avcodec.c
index 4abe9f8b..0b00f1f7 100644
--- a/src/io/source_avcodec.c
+++ b/src/io/source_avcodec.c
@@ -347,8 +347,13 @@ void aubio_source_avcodec_reset_resampler(aubio_source_avcodec_t * s)
     av_channel_layout_default(&input_layout, s->input_channels);
     av_channel_layout_default(&output_layout, s->input_channels);
 
+#if LIBSWRESAMPLE_VERSION_INT < AV_VERSION_INT (4, 5, 100)
+    av_opt_set_chlayout(avr, "in_channel_layout",  &input_layout,        0);
+    av_opt_set_chlayout(avr, "out_channel_layout", &output_layout,       0);
+#else
     av_opt_set_chlayout(avr, "in_chlayout",  &input_layout,        0);
     av_opt_set_chlayout(avr, "out_chlayout", &output_layout,       0);
+#endif
 #else
     int64_t input_layout = av_get_default_channel_layout(s->input_channels);
     int64_t output_layout = av_get_default_channel_layout(s->input_channels);
